ttnn/cpp/ttnn/operations/data_movement/tilize_with_val_padding/device/tilize_with_val_padding_program_factory.cpp

```diff
@@ -444,8 +471,17 @@ operation::ProgramWithCallbacks tilize_with_val_padding_multi_core_sharded(
     auto tilize_kernel_id = CreateKernel(
         program, "ttnn/cpp/ttnn/deprecated/tt_dnn/kernels/compute/tilize.cpp", all_cores, ComputeConfig{.compile_args = compute_args});
 
-    bfloat16 bfloat_pad_value = bfloat16(pad_value);
-    uint32_t packed_pad_value = pack_two_bfloat16_into_uint32({bfloat_pad_value, bfloat_pad_value});
+    uint32_t packed_pad_value;
+    if (a.get_dtype() == DataType::BFLOAT16) {
```

Comment:
in general I would add support for Float32 here I guess you will need it soon enough